### PR TITLE
added InfoVAE implementation for generating brain scans based on OASIS

### PR DESCRIPTION
**Pull request by 1shz**
Name: Ishraque Zahin
SID: 45129256

Hi, I've created an InfoVAE implementation to expand the PatternFlow collection.

This implementation uses a simple auto encoder model that uses convolutional layers to scale down/up the data while inferring structure and a few fully connected dense layers to map to/from the latent space. The size of the dense layers are scaled according to the input image size (256 by default). The decoder is symmetrical to the encoder. The MMD loss function used for the encoder is adapted from [Shengjia Zhao's InfoVAE implementation](https://github.com/ShengjiaZhao/MMD-Variational-Autoencoder/). The decoder uses a MSE (Mean Squared Error) loss to minimise the difference between the reconstruction and original image. The model is defined in `model.py`.

We use the [OASIS MRI scan dataset](https://learn.uq.edu.au/webapps/blackboard/content/listContent.jsp?course_id=_132224_1&content_id=_5540995_1) to demonstrate the generation of images that look like brain scans. Data imports are handled by `data.py`

The script for the training loop, plotting images, and random latent encoding generation have been adapted from the Tensorflow CVAE tutorial, and are used in the driver notebook/script `run.ipynb`.

Please suggest improvements to documentation (e.g README.md).

I think the implementation works well and is parameterised on latent_dimensions and the input size, but perhaps we can have better parameters? 
Also, can I get bonus points for my tensorflow purity? I didn't import numpy anywhere! :P

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
@shakes76  are students allows to (re)-implements info-vae?

**Comment by shakes76** [comment]
@SiyuLiu0329 Yes they can from the OASIS or AKOA datasets, which was not shown

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required.**
**Add more comments to driver. Notebooks have markdown sections for you to explain what's going on, so you should make use of it.**
No Proper Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Comment by Ishraque99** [comment]
@shakes76 updated initial PR description and updated documentation in driver script
